tasks/linting-diff/lintingDiff.mts

```diff
@@ -0,0 +1,117 @@
+import { $, glob, spinner, fs, path } from 'zx'
+import { cpus } from 'node:os'
+
+async function getConfigsForFiles(files: string[]) {
+  const configs = new Map<string, any>()
+
+  // Batch these in groups because running all of them in parallel is too many
+  // processes to spawn at once
+  const batchSize = cpus().length
+  for (let i = 0; i < files.length; i += batchSize) {
+    const batch = files.slice(i, i + batchSize)
+    process.stdout.clearLine(0)
+    process.stdout.cursorTo(0)
+    process.stdout.write(`Progress: ${Math.min(i + batchSize, files.length)}/${files.length}`)
+    const promises = batch.map(async (file) => {
+      const { stdout } = await $`yarn eslint --print-config ${file}`
+      configs.set(file, JSON.parse(stdout))
+    })
+    await Promise.all(promises)
+  }
+  process.stdout.write('\n')
+
+  return configs
+}
+
+async function main() {
+  // We need to be running commands within a redwood project
+  if (!process.env.RWJS_CWD) {
+    throw new Error('RWJS_CWD is not set')
+  }
+  $.verbose = !!process.env.VERBOSE
+  $.cwd = process.env.RWJS_CWD
+
+  // Get all the files in the project
+  const files = await glob('**/*.*', {
+    ignore: ['node_modules', 'dist', 'yarn.lock'],
+    cwd: $.cwd,
+  })
+
+  // We support a `--sample` flag to only run on a sample of the files
+  if (process.argv.includes('--sample')) {
+    const sampleSize = cpus().length
+    console.log(`Running on a sample of ${sampleSize} files`)
+    files.sort(() => Math.random() - 0.5).splice(sampleSize)
+  }
```

Comment:
Had to put on my imaginary physicist hat for this

Reply:
I put my lazy person hat on for that one because I couldn't be bothered waiting for 138 files every time I was testing this haha.